packages/message-predicates/contract-message-predicate/src/lib.rs

```diff
@@ -1,24 +1,62 @@
-mod predicate_asm;
-mod script_asm;
-
 use fuel_tx::{ConsensusParameters, Input};
-use sha2::{Digest, Sha256};
 
-// Make the script and predicate bytecode public
-pub use predicate_asm::bytecode as predicate_bytecode;
-pub use script_asm::bytecode as script_bytecode;
+pub const SCRIPT_HASH: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/out/contract_message_script_hash.bin"
+));
+pub const SCRIPT_ASM: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/out/contract_message_script.bin"
+));
+pub const PREDICATE_BYTECODE: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/out/contract_message_predicate.bin"
+));
+pub const DEFAULT_PREDICATE_ROOT: &[u8] = include_bytes!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/out/contract_message_predicate_default_root.bin"
+));
+
+// Gets the bytecode for the message-to-contract script
+pub fn script_bytecode() -> Vec<u8> {
+    SCRIPT_ASM.to_vec()
+}
+
+// Gets the bytecode for the message-to-contract predicate
+pub fn predicate_bytecode() -> Vec<u8> {
+    PREDICATE_BYTECODE.to_vec()
+}
 
 // Gets the hash of the message-to-contract script
 pub fn script_hash() -> [u8; 32] {
-    let script = script_asm::bytecode();
-    let mut script_hasher = Sha256::new();
-    script_hasher.update(script);
-    script_hasher.finalize().into()
+    SCRIPT_HASH
```

Comment:
My understanding is that this used to do a sha256 hash but now it takes the bytes from the binary and attempts to convert it into an array. Is hashing done somewhere else?
I'm unsure of what `script_asm::bytecode()` used to do.

Reply:
Looks like the previous code has been moved to `build.rs`

Reply:
Yeah this is just an optimization to avoid running sha256 as often at runtime since this value never changes.